### PR TITLE
rough draft of singularity in htc guide

### PR DESCRIPTION
**Pull request by ChristinaLK**
Needs review for
- whether the instructions are accurate @jasoncpatton 
- the recommendations for transfer/squid/staging are what we want @lmichael107 @bbockelm 
- formatting and un-broken links @CannonLock

### CODE REVIEW

File: uw-research-computing/singularity-htc.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+1. Then, save the Docker container to a Singularity image. You'll need the name of 
+the Docker image on DockerHub or (the link to it at another location) and you will 
+provide the name you want to use for the saved Singularity image file: 
+	``` 
+	[alice@build]$ singularity build /software/alice/name.simg docker://user/image:version
+	```
+	{:.term}
+	
+	For example, if user \"Alice\" wanted to use the [\"Fenics\" container
+	provided on DockerHub](https://hub.docker.com/r/fenicsproject/stable),
```
Comment (CannonLock) [inline uw-research-computing/singularity-htc.md:43]: This link is broken.

**Review by CannonLock — CHANGES_REQUESTED** [review]

File: uw-research-computing/singularity-htc.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+variable that prevents download of the Docker container. 
+	```
+	[alice@build]$ unset HTTPS_PROXY
+	```
+
+1. Then, save the Docker container to a Singularity image. You'll need the name of 
+the Docker image on DockerHub or (the link to it at another location) and you will 
+provide the name you want to use for the saved Singularity image file: 
+	``` 
+	[alice@build]$ singularity build /software/alice/name.simg docker://user/image:version
```
Comment (ChristinaLK) [inline uw-research-computing/singularity-htc.md:38]:
```suggestion
	[alice@build]$ singularity build /software/alice/name.sif docker://user/image:version
```

File: uw-research-computing/singularity-htc.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	```
+	{:.term}
+	
+	For example, if user \"Alice\" wanted to use the [\"Fenics\" container
+	provided on DockerHub](https://hub.docker.com/r/fenicsproject/stable),
+	and save it to a file named `fenics.simg`, she would run:
```
Comment (ChristinaLK) [inline uw-research-computing/singularity-htc.md:44]:
```suggestion
	and save it to a file named `fenics.sif`, she would run:
```

File: uw-research-computing/singularity-htc.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+	``` 
+	[alice@build]$ singularity build /software/alice/fenics.simg docker://fenicsproject/stable:latest
```
Comment (ChristinaLK) [inline uw-research-computing/singularity-htc.md:47]:
```suggestion
	[alice@build]$ singularity build /software/alice/fenics.sif docker://fenicsproject/stable:latest
```

File: uw-research-computing/singularity-htc.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+- **For Image Files < 100MB use HTCondor File Transfer**
+
+If the Singularity image file is less than 100MB, it can be copied 
+to the job from the home directory using HTCondor file transfer. 
+
+The submit file options should look like this: 
+
+``` {.sub}
+transfer_input_files = image.simg
```
Comment (ChristinaLK) [inline uw-research-computing/singularity-htc.md:71]:
```suggestion
transfer_input_files = image.sif
```

File: uw-research-computing/singularity-htc.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
++SingularityImage = "./image.simg"
```
Comment (ChristinaLK) [inline uw-research-computing/singularity-htc.md:72]:
```suggestion
+SingularityImage = "./image.sif"
```

File: uw-research-computing/singularity-htc.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+- **For Image Files < 1GB use SQUID**
+
+If the Singularity image file is larger than 100MB but less than 1GB, it can be 
+placed on our SQUID web server and downloaded into the job. Follow the instructions 
+on this page to gain access to SQUID and upload your data: [File Availability with SQUID](file-avail-squid)
+
+Once the image is staged in SQUID, use the following submit file options: 
+
+``` {.sub}
+transfer_input_files = http://proxy.chtc.wisc.edu/SQUID/user/image.simg
```
Comment (ChristinaLK) [inline uw-research-computing/singularity-htc.md:86]:
```suggestion
transfer_input_files = http://proxy.chtc.wisc.edu/SQUID/user/image.sif
```

File: uw-research-computing/singularity-htc.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
++SingularityImage = "./image.simg"
```
Comment (ChristinaLK) [inline uw-research-computing/singularity-htc.md:87]:
```suggestion
+SingularityImage = "./image.sif"
```

File: uw-research-computing/singularity-htc.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+- **For Image Files > 1GB use Shared Software Directory**
+
+If the Singularity image file is larger than 1GB, request a shared `/software` directory 
+from the CHTC facilitators by emailing chtc@cs.wisc.edu. Note that the image should NOT be transferred to the job via `transfer_input_files`.
+
+Once the image is placed in `/software`, use the following submit file options: 
+
+``` {.sub}
++SingularityImage = "/software/user/image.simg"
```
Comment (ChristinaLK) [inline uw-research-computing/singularity-htc.md:100]:
```suggestion
+SingularityImage = "/software/user/image.sif"
```

**Review by ChristinaLK — COMMENTED** [review]
changing image file extension

**Comment by ChristinaLK** [comment]
This should now be ready to merge except that it's out of date because of container universe. 😭 
Will revisit after new years.

**Comment by aowen-uwmad** [comment]
This PR is obsolete after #582

**Pull request closed without merging**